src/std2_type_traits.hpp

```diff
@@ -0,0 +1,34 @@
+/** ==========================================================================
+ * 2013 This is PUBLIC DOMAIN to use at your own risk and comes
+ * with no warranties. This code is yours to share, use and modify with no
+ * strings attached and no restrictions or obligations.
+ * 
+ * These type traits will be in C++17.
+ * 
+ * PUBLIC DOMAIN and NOT under copywrite protection.
+ *
+ * ********************************************* */
+
+#ifndef STD2_TYPE_TRAITS_HPP_
```

Comment:
Can you replace this for #pragma once instead?

Reply:
You already done this. resolving :)